### PR TITLE
[NCL-5663] Update Keycloak denepdencies

### PR DESCRIPTION
**Pull request by janinko**

### CODE REVIEW

File: causeway-client/pom.xml [hunk 1]
```diff
@@ -65,9 +65,12 @@
...
     <dependency>
-      <groupId>org.keycloak</groupId>
-      <artifactId>keycloak-core</artifactId>
-      <scope>test</scope>
+        <groupId>com.fasterxml.jackson.core</groupId>
+        <artifactId>jackson-annotations</artifactId>
+    </dependency>
+    <dependency>
+        <groupId>com.fasterxml.jackson.datatype</groupId>
+        <artifactId>jackson-datatype-jsr310</artifactId>
```
Comment (pkocandr) [inline causeway-client/pom.xml:73]: How are these 2 deps related to the keycloak version change?

File: pom.xml [hunk 2]
```diff
@@ -224,6 +224,13 @@
         <type>pom</type>
         <scope>import</scope>
       </dependency>
+      <dependency>
+        <groupId>com.redhat.bom.rh-sso</groupId>
+        <artifactId>rh-sso-adapter-bom</artifactId>
+        <version>7.4.0.GA</version>
```
Comment (pkocandr) [inline pom.xml:230]: Does the version match the one in EAP 7.2.2.GA? We are not on 7.3 yet...

Comment (pkocandr) [reply pom.xml:230]: Also I would add a comment like "if changed version.keycloak value must be changed accordingly" to avoid the mess in the future.

File: causeway-client/pom.xml (hunk 1 above)
Comment (janinko) [reply causeway-client/pom.xml:73]: I removed keycloak from causeway-client, which was sneaking these dependencies here.

File: pom.xml (hunk 2 above)
Comment (janinko) [reply pom.xml:230]: We are going to move to EAP 7.3 "soon". Show I merge it anyway or wait for the move to 7.3 EAP?
